### DISCUSSION TITLE
workflow async execute

### DISCUSSION THREAD

**Discussion by nicedream7758** [langgenius/dify#4489] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

now, we use  service api  post to /workflow/run, but it is a  sync api,  some times it cost a lot of time,  will  reduce system performance

### 2. Describe the feature you'd like to see

maybe can add another interface, like /workflow/run/async,   run  workflow on bakend

### 3. How will this feature improve your workflow or experience?

it can be quickly return when exec any long time workflow, reduce the request time,  actually  we did this in local env.

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by changjiangbailang** [comment]
+1

**Comment by LyleLiu666** [comment]
This feature is very useful and can greatly improve work efficiency in production environments

**Comment by f1ynng8** [comment]
I'm using a work-arount like this to have the async performance.
1. add timeout=1 parameter to the api invoker like `response = requests.post(api_url, headers=headers, json=data, timeout=1)`, remember to do the requests.exceptions.Timeout process.
2. add a HTTP node in the dify workflow to send the result to a callback url of the invoker.
